### ISSUE TITLE
Try to Populate Local Workspace: Error "Cannot read property 'indexOf' of null"

### ISSUE THREAD

**Issue by SLathropDP** [readysitego/spgo#16]
Attempting to populate local workspace from SharePoint Online (O365).

SPGo.json config:

```
{
  "sourceDirectory": "src",
  "sharePointSiteUrl": "https://TENANT.sharepoint.com",
  "publishingScope": "Major",
  "authenticationType": "Digest",
  "remoteFolders": ["/SiteAssets/", "/_catalogs/"]
}
```

Log shows:

```
SPGo enabled.
Starting File Synchronization...
================================     ERROR     ================================

Cannot read property 'indexOf' of null
===============================================================================
```

**Comment by sg-chrishasz** [comment]
Hello @SLathropDP 
I'm reviewing this issue now. I'll post an update once I've been able to reproduce and have an idea of next steps.

Thanks for using SPGo!
-Chris

**Comment by sg-chrishasz** [comment]
@SLathropDP -
I think there are two issues occurring here:

**Issue 1:** 
The folder set <site>/_catalogs/* are managed paths, which point directly to the file system in SharePoint web apps. Due to this, the path /_catalogs/ does not actually exist (if you navigate to this url, you will get a 400 error). Per SharePoint's architecture, you will need to specify each subfolder directly in your SPGO.json config file. The good news is that SPGo will work if you update the remoteFolders property to specify each subdirectly directly. I've provided a sample config file below to better illustrate this.

***example:***
```{
  "sourceDirectory": "src",
  "sharePointSiteUrl": "https://TENANT.sharepoint.com",
  "publishingScope": "Major",
  "authenticationType": "Digest",
  "remoteFolders": ["/SiteAssets/", "/_catalogs/masterpage/", "/_catalogs/wp/"]
}```

**Issue 2:** 
SPGo was not correctly handling the 400 response from SharePoint, causing the error you experience. I've updated how SPGo handles file download errors to prevent this from occurring.

I'll release the fix as part of 0.10.2 later today.

Thanks,
Chris
